internal/entity/dto/v1/domain.go

```diff
@@ -12,3 +20,12 @@ type Domain struct {
 	TenantID                      string    `json:"tenantId" example:"abc123"`
 	Version                       string    `json:"version,omitempty" example:"1.0.0"`
 }
+
+// ValidateAlphaNumHyphenUnderscore validates that a field contains only alphanumeric characters, hyphens, and underscores.
+func ValidateAlphaNumHyphenUnderscore(fl validator.FieldLevel) bool {
```

Comment:
Please use compiled regex at package level, to avoid repeated compilations like
```
var alphanumericWithSpecialCharsRegex = regexp.MustCompile(`^[a-zA-Z0-9_-]+$`)

func ProfileNameValidator(fl validator.FieldLevel) bool {
	return alphanumHyphenUnderscoreRegex.MatchString(fl.Field().String())
}
```
This validator can be reused for other profile names (wireless profiles, IEEE 802.1x profiles, CIRA profiles, etc.)
that need to match the alphanumeric with hyphens and underscores pattern.